test/framework/src/main/java/org/opensearch/test/OpenSearchIntegTestCase.java

```diff
@@ -2349,4 +2358,88 @@ protected ClusterState getClusterState() {
+                        assertBusy(() -> {
+                            for (IndexShardRoutingTable shardRoutingTable : indexRoutingTable) {
+                                final ShardRouting primaryRouting = shardRoutingTable.primaryShard();
+                                if (primaryRouting.state().toString().equals("STARTED")) {
+                                    final String indexName = primaryRouting.getIndexName();
+                                    if (isSegmentReplicationEnabledForIndex(indexName)) {
+                                        final List<ShardRouting> replicaRouting = shardRoutingTable.replicaShards();
+                                        final IndexShard primaryShard = getIndexShard(primaryRouting, indexName);
+                                        final Map<String, StoreFileMetadata> primarySegmentMetadata = primaryShard.getSegmentMetadataMap();
+                                        for (ShardRouting replica : replicaRouting) {
+                                            if (replica.state().toString().equals("STARTED")) {
+                                                IndexShard replicaShard = getIndexShard(replica, indexName);
+                                                final Store.RecoveryDiff recoveryDiff = Store.segmentReplicationDiff(
+                                                    primarySegmentMetadata,
+                                                    replicaShard.getSegmentMetadataMap()
+                                                );
+                                                if (recoveryDiff.missing.isEmpty() == false || recoveryDiff.different.isEmpty() == false) {
+                                                    fail(
+                                                        "Expected no missing or different segments between primary and replica but diff was missing: "
+                                                            + recoveryDiff.missing
+                                                            + " Different: "
+                                                            + recoveryDiff.different
+                                                            + " Primary Replication Checkpoint : "
+                                                            + primaryShard.getLatestReplicationCheckpoint()
+                                                            + " Replica Replication Checkpoint: "
+                                                            + replicaShard.getLatestReplicationCheckpoint()
+                                                    );
+                                                }
+
+                                                // calls to readCommit will fail if a valid commit point and all its segments are not in the
+                                                // store.
+                                                replicaShard.store().readLastCommittedSegmentsInfo();
```

Comment:
I don't think this is required for the purpose of waiting until its up to date.

Reply:
Done, will remove in next commit.